Dependabot Docs: update guidance-for-the-configuration-of-private-registries (#28884)
github/docs: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md

- [Azure Artifacts](#azure-artifacts)
- [{% data variables.product.prodname_registry %} registry](#data-variablesproductprodname_registry--registry)
- [Nexus](#nexus)
- [ProGet](#proget)

## Configuring package managers

### Bundler

> Supported by Artifactory, Artifacts, {% data variables.product.prodname_registry %} registry, Nexus, and ProGet.

Reviewer: @whatatripp it doesn't seem like ProGet supports Bundler from reading the docs right?

```suggestion
```

Author: Yes (Rubygems) - added an example  a bottom :)

### Python

> Supported by Artifactory, Azure Artifacts, Nexus, and ProGet. The {% data variables.product.prodname_registry %} registry is not supported.

Reviewer: @whatatripp it doesn't seem like ProGet supports Python packages from reading the docs right?

```suggestion
```

Author: // same (Python feeds) - example added :)

For more information, see [Securing Nexus Repository Manager](https://help.sonatype.com/repomanager3/planning-your-implementation/securing-nexus-repository-manager) in the Sonatype documentation.

  Registries can be proxied to reach out to a public registry in case a dependency is not available in the private regstry.{% ifversion dependabot-private-registries %} However, you may want {% data variables.product.prodname_dependabot %} to only access the private registry and not access the public regsitry at all. For more information, see [Quick Start Guide - Proxying Maven and NPM](https://help.sonatype.com/repomanager3/planning-your-implementation/quick-start-guide---proxying-maven-and-npm)" in the Sonatype documentation, and "[AUTOTITLE](/code-security/dependabot/working-with-dependabot/removing-dependabot-access-to-public-registries)."{% endif %}

### ProGet

> For information about ProGet and instructions on how to configure {% data variables.product.prodname_dependabot %} to work with feeds in ProGet, see [the ProGet documentation](https://docs.inedo.com/docs/proget-overview).

Reviewer 2: Minor
```suggestion
For information about ProGet and instructions on how to configure {% data variables.product.prodname_dependabot %} to work with feeds in ProGet, see the [ProGet documentation](https://docs.inedo.com/docs/proget-overview).
```

Example of ProGet registry for a NuGet feed:

{% raw %}

```yaml
registries:
  proget-nuget-feed:
    type: nuget-feed
    url: https://proget.corp.local/nuget/MyNuGetFeed/v3/index.json
    token: ${{secrets.PROGET_APK_KEY}}
```

{% endraw %}

> 

Reviewer 2: Let's follow the same structure as for the other private registry hosts.
```suggestion

#### Notes

```

> The `token` should be an API Key that with access to view packages; see [API Access and API Keys](https://docs.inedo.com/docs/buildmaster-administration-security-api-keys).

Reviewer 2: ```suggestion
The `token` should be an API Key with access to view packages. For more information, see [API Access and API Keys](https://docs.inedo.com/docs/buildmaster-administration-security-api-keys) in the ProGet documentation.
```

For more information, see [Securing Nexus Repository Manager](https://help.sonatype.com/repomanager3/planning-your-implementation/securing-nexus-repository-manager) in the Sonatype documentation.

  Registries can be proxied to reach out to a public registry in case a dependency is not available in the private regstry.{% ifversion dependabot-private-registries %} However, you may want {% data variables.product.prodname_dependabot %} to only access the private registry and not access the public regsitry at all. For more information, see [Quick Start Guide - Proxying Maven and NPM](https://help.sonatype.com/repomanager3/planning-your-implementation/quick-start-guide---proxying-maven-and-npm)" in the Sonatype documentation, and "[AUTOTITLE](/code-security/dependabot/working-with-dependabot/removing-dependabot-access-to-public-registries)."{% endif %}

### ProGet

For information about ProGet and instructions on how to configure {% data variables.product.prodname_dependabot %} to work with feeds in ProGet, see the [ProGet documentation](https://docs.inedo.com/docs/proget-overview).

> Example of ProGet registry for a NuGet feed:

Reviewer 2: ```suggestion
Example of ProGet registry configuration for a NuGet feed:
```

{% raw %}

```yaml
registries:
  proget-nuget-feed:
    type: nuget-feed
    url: https://proget.corp.local/nuget/MyNuGetFeed/v3/index.json
    token: ${{secrets.PROGET_APK_KEY}}
```

{% endraw %}

> Example of ProGet registry for Bundler (rubygems):

Reviewer 2: ```suggestion
Example of ProGet registry configuration for Bundler (rubygems):
```

{% raw %}

```yaml
registries:
  proget-gems-feed:
    type: rubygems-server
    url: https://proget.corp.local/rubygems/MyRubygemsFeed
    token: ${{secrets.PROGET_APK_KEY}}
```

{% endraw %}

> Example of ProGet registry for Python (PyPI):

Reviewer 2: ```suggestion
Example of ProGet registry configuration for Python (PyPI):
```
